Migrate Harbor UI E2E tests from Robot + Selenium to Playwright

bupd opened the pull request:
Tracking Issue: https://github.com/goharbor/harbor/issues/22134

bupd committed:
feat: migrate ui e2e from robot to playwright

Signed-off-by: bupd <bupdprasanth@gmail.com>

MinerYang reviewed (changes requested):
Hi @bupd 
Please help to update the proposal per this comment and the discussion we had last month.
https://github.com/goharbor/community/pull/268#issuecomment-3461261539

Best,
Miner

bupd committed:
Enhance Playwright migration proposal details

Expand on the benefits and features of using Playwright for UI testing, including portability, containerization, and improved debugging capabilities. Outline goals and non-goals for the migration process from Robot tests to Playwright.

Signed-off-by: Prasanth Baskar <bupdprasanth@gmail.com>

bupd commented:
@MinerYang @stonezdj @reasonerjt  All concerns have been addressed.

MinerYang commented on proposals/new/ui-e2e-migrate-playwright.md:
```diff
@@ -0,0 +1,106 @@
+
+### Portability & Containerization
+- Playwright works consistently across Jenkins, GitHub Actions, GitLab, air-gapped networks, and plain shell environments.
+- A single container image can include Playwright browsers + required CLIs (docker, helm, cosign, notation, oras, curl, jq).
```
Thanks, And all these details  would be better to presents in the proposal.

bupd committed:
feat: update playwright options and examples

Signed-off-by: bupd <bupdprasanth@gmail.com>

bupd committed:
docs: address technical concerns in Playwright migration proposal

Add Technical Considerations section covering:
- Process orchestration and parallelism handling
- Credential safety in CI environments
- SSH operations support
- Reporting comparison with Robot Framework

Clarify proposal scope to UI E2E tests only, with infrastructure tests remaining in Robot Framework.

Remove bold formatting per style guide.

Signed-off-by: bupd <bupdprasanth@gmail.com>

bupd commented on proposals/new/ui-e2e-migrate-playwright.md:
```diff
@@ -0,0 +1,106 @@
+- It supports fixtures and helper modules. https://playwright.dev/docs/test-fixtures
+- Skipping, tagging, grouping, and environment-driven execution are built-in https://playwright.dev/docs/test-annotations
+- Playwright can run CLI operations via Node's `child_process` reliably (docker, helm, oras, cosign, notation).
+- Reports, videos, traces, retries, and serial execution already meet & exceed current quality standards.
```
- https://github.com/goharbor/harbor/pull/22591 pr covering bigger testcase with securityhub interacting with harbor UI, trivy etc. this PR also showcases full build and test workflow.

vg006 replied:
@MinerYang

I'm currently migrating all the test cases in the [`Common.robot`](https://github.com/goharbor/harbor/blob/c694f5cee9c97058bffd344a0914d9fe20250297/tests/robot-cases/Group1-Nightly/Common.robot) file. *You can check the progress in this PR https://github.com/goharbor/harbor/pull/22630*

The mentioned [Test Case - Cosign And Cosign Deployment Security Policy](https://github.com/goharbor/harbor/blob/c694f5cee9c97058bffd344a0914d9fe20250297/tests/robot-cases/Group1-Nightly/Common.robot#L796) perform signing the image in the repository using `cosign`. But the signatures by `cosign` (*version 3.0+*) aren't properly recognized by Harbor. This issue is explained in detail and has been resolved at the same in this PR https://github.com/goharbor/harbor/pull/22628. Now this fix is planned to be released in the upcoming version `2.15` of Harbor.

I have migrated this test case to **Playwright**, but need to test and verify it against the newer version, before committing them. So the changes are in my local branch and haven't pushed yet.

Thus the successful migration of this particular test case will be in halt, until the newer version is released. 
Thank you.

bupd committed:
docs: add running commands and keyword-driven execution Q&A

Update Dockerfile to Ubuntu 20.04-based image matching Robot Framework structure.
Add comprehensive Q&A section answering:
- How running commands look like (Docker and direct execution)
- Keyword-driven execution support with real examples

Include references to PRs #22462 and #22591 showing actual implementation.

Signed-off-by: bupd <bupdprasanth@gmail.com>

bupd committed:
docs: add HTML report, video recording, and interactive debug screenshots

Add three screenshots demonstrating Playwright reporting capabilities:
- HTML test report showing all tests passing
- Video recording feature in test reports
- Interactive debug mode with Playwright Inspector

Signed-off-by: bupd <bupdprasanth@gmail.com>

bupd committed:
Merge branch 'main' into migrate-e2e-proposal

bupd committed:
Merge branch 'main' into migrate-e2e-proposal

wy65701436 reviewed (approved):
lgtm

bupd committed:
Merge branch 'main' into migrate-e2e-proposal
